### PR TITLE
Update Kinesis Mock to 0.3.0

### PR DESCRIPTION
**Pull request by etspaceman**
See https://github.com/etspaceman/kinesis-mock/releases/tag/0.3.0

This adds the UpdateStreamMode API, as well as updates to a few other request/response bodies with information from the stream mode. 

Also see https://docs.aws.amazon.com/kinesis/latest/APIReference/API_UpdateStreamMode.html

### CODE REVIEW

File: localstack/config.py [hunk 1]
```diff
@@ -552,6 +552,11 @@ def in_docker():
 # limit in which to kinesis-mock will start throwing exceptions
 KINESIS_SHARD_LIMIT = os.environ.get("KINESIS_SHARD_LIMIT", "").strip() or "100"
 
+# limit in which to kinesis-mock will start throwing exceptions
+KINESIS_ON_DEMAND_STREAM_COUNT_LIMIT = os.environ.get(
+    "KINESIS_ON_DEMAND_STREAM_COUNT_LIMIT", ""
```
Comment (thrau) [inline localstack/config.py:557]: for the CLI to pass through the env var correctly to the container, it needs to be whitelisted in the `CONFIG_ENV_VARS` further down in `config.py`

**Review by thrau — CHANGES_REQUESTED** [review]
Thanks Eric @etspaceman! Great to see another kinesis-mock release :-)

Just one minor change necessary for the merge.

File: localstack/config.py (hunk 1 above)
Comment (etspaceman) [reply localstack/config.py]: Fixed

**Review by thrau — APPROVED** [review]
Thanks! LGTM

**Pull request merged**